sourcetracker/tests/test_sourcetracker.py

```diff
@@ -9,96 +9,209 @@
+                               columns=self.columns)
+        data = np.random.randint(0, 10, size=200).reshape(50, 4)
+        sinks = pd.DataFrame(data.astype(np.int32),
+                             index=['s%s' % i for i in range(50)],
+                             columns=self.columns)
+        obs_sources, obs_sinks = validate_gibbs_input(sources, sinks)
+        pd.util.testing.assert_frame_equal(obs_sources, sources)
+        pd.util.testing.assert_frame_equal(obs_sinks, sinks)
+
+    def test_columns_non_identical(self):
+        # Columns are not identical, error expected.
+        data = np.random.randint(0, 10, size=20).reshape(5, 4)
+        sources = pd.DataFrame(data.astype(np.int32),
+                               index=self.index,
+                               columns=self.columns)
+        data = np.random.randint(0, 10, size=200).reshape(50, 4)
+        sinks = pd.DataFrame(data.astype(np.int32),
+                             index=['s%s' % i for i in range(50)],
+                             columns=['feature%s' % i for i in range(4)])
+        self.assertRaises(ValueError, validate_gibbs_input, sources, sinks)
+
+
+class TestValidateGibbsParams(TestCase):
+
+    def test_acceptable_inputs(self):
+        # All values acceptable, expect no errors.
+        alpha1 = .001
+        alpha2 = .1
+        beta = 10
+        restarts = 10
+        draws_per_restart = 1
+        burnin = 100
+        delay = 1
+        self.assertTrue(validate_gibbs_parameters(alpha1, alpha2, beta,
+                        restarts, draws_per_restart, burnin, delay))
+
+        alpha1 = alpha2 = beta = 0
+        self.assertTrue(validate_gibbs_parameters(alpha1, alpha2, beta,
+                        restarts, draws_per_restart, burnin, delay))
+
+    def test_not_acceptable_inputs(self):
+        # One of the float params is negative.
+        alpha1 = -.001
+        alpha2 = .1
+        beta = 10
+        restarts = 10
+        draws_per_restart = 1
+        burnin = 100
+        delay = 1
+        self.assertFalse(validate_gibbs_parameters(alpha1, alpha2, beta,
+                         restarts, draws_per_restart, burnin, delay))
 
-        obs = biom_to_df(Table(data, oids, sids))
-        pd.util.testing.assert_frame_equal(obs, self.exp)
+        # One of the int params is 0.
+        alpha1 = .001
+        restarts = 0
+        self.assertFalse(validate_gibbs_parameters(alpha1, alpha2, beta,
+                         restarts, draws_per_restart, burnin, delay))
 
-        # Because the biom table and dataframe are transposed with respect to
-        # one another, we used data[3, 0] and exp.iloc[0, 3] to refer to the
-        # same value in the different tables.
-        data[3, 0] = 4.512
-        obs = biom_to_df(Table(data, oids, sids))
-        exp = self.exp.copy()
-        exp.iloc[0, 3] = 5
-        pd.util.testing.assert_frame_equal(obs, exp)
+        # One of the int params is a float.
```

Comment:
Have you tested when one of the float params is an int, or do you just want to accept that? What if they're strings?